src/lido/layout.ts

```diff
@@ -0,0 +1,79 @@
+import { publicKey, struct, u64, u8, u32, array, vec } from "@project-serum/borsh";
+
+const METRICS_LAYOUT = struct(
+  [
+    u64("feeTreasurySolTotal"),
+    u64("feeValidationSolTotal"),
+    u64("feeDeveloperSolTotal"),
+    u64("stSolAppreciationTotal"),
+    u64("feeTreasuryStSolTotal"),
+    u64("feeValidationStSolTotal"),
+    u64("feeDeveloperStSolTotal"),
+    struct([array(u64(), 12, "counts"), u64("total")], "depositAmount"),
+    struct([u64("totalStSolAmount"), u64("totalSolAmount"), u64("count")], "withdrawAmount"),
+  ],
+  "metrics"
+);
+
+/**
+ * Seed range with a begin and end, both u64
+ *
+ * @param property optional name for the seed range
+ * @returns (named) struct containing a seed range
+ */
+function seedRange(property?: string | undefined): struct {
+  return struct([u64("begin"), u64("end")], property);
+}
+
+/**
+ * Layout of a single validator
+ */
+const VALIDATOR_LAYOUT = struct(
+  [
+    u64("feeCredit"),
+    publicKey("feeAddress"),
+    seedRange("stakeSeeds"),
+    seedRange("unstakeSeeds"),
```

Comment:
Since `seedRange` will only be used right here, it's a bit redundant imo.
Replace it with below is fine.

```
struct([u64("begin"), u64("end")], "stakeSeeds");
struct([u64("begin"), u64("end")], "unstakeSeeds");
```

Reply:
Yeah, I agree with you on this. I think I got a little too excited with separating everything XD.

Reply:
Anyways, I've updated it now :)